runtime/src/anchor.rs

```diff
@@ -194,6 +209,44 @@ decl_module! {
             }
             Ok(())
         }
+
+        /// Initiates eviction of expired anchors. Since anchors are stored on a child trie indexed by
+        /// their eviction date, what this function does is to remove those child tries which has
+        /// date_represented_by_root < current_date. Additionally it needs to take care of indexes
+        /// created for accessing anchors, eg: to find an anchor given an id.
+        pub fn evict_anchors(origin) -> Result {
+            ensure_signed(origin)?;
+            let current_timestamp = <timestamp::Module<T>>::get();
+
+            // get the today counting epoch, so that we can remove the corresponding child trie
+            let today_in_days_from_epoch = TryInto::<u64>::try_into(current_timestamp)
+                .map(common::get_days_since_epoch)
+                .map_err(|_e| "Can not convert timestamp to u64")
+                .unwrap();
+            let evict_date = LatestEvictedDate::get();
+
+            // remove child tries starting from day next to last evicted day
+            let evicted_trie_count = (evict_date + 1..today_in_days_from_epoch)
+                .map(|day| {(day, common::generate_child_storage_key(day))})
+                // store the root of child trie for the day on chain before eviction. Checks if it
+                // exists before hand to ensure that it doesn't overwrite a root.
+                .map(|(day, key)| {
+                    if !EvictedAnchorRoots::exists(day) {
+                        EvictedAnchorRoots::insert(day, runtime_io::child_storage_root(&key));
+                    }
+                    key
+                })
+                .map(|key| runtime_io::kill_child_storage(&key))
+                .count();
+            // store yesterday as the last day of eviction
+            let yesterday = today_in_days_from_epoch - 1;
+            LatestEvictedDate::put(yesterday);
+            let evicted_anchor_indexes_count = Self::remove_anchor_indexes(yesterday);
+
+            // TODO emit an event for this so that the process which triggers can know how many anchor indexes got purged
```

Comment:
I'd love to have an event that gets emitted when a bucket got evicted.
not sure though for what we would need the event of "how many" anchors got evicted. what do you think?